packages/runtime/src/store/editor.ts

```diff
@@ -13,12 +14,14 @@ export interface ScrollPosition {
   left: number;
 }
 
-export type EditorDocuments = Record<string, EditorDocument>;
+export type EditorDocuments = Record<string, EditorDocument | undefined>;
 
 export class EditorStore {
+  editorConfig = atom<EditorConfig>(new EditorConfig());
   selectedFile = atom<string | undefined>();
   documents = map<EditorDocuments>({});
 
+  files = computed(this.documents, (documents) => Object.keys(documents).sort());
```

Comment:
We should provide a custom sort that always sort folders first which is what we get in vscode:

```suggestion
  files = computed(this.documents, (documents) => Object.keys(documents).sort((a, b) => {
    const segmentsA = a.split('/');
    const segmentsB = b.split('/');

    const minLength = Math.min(segmentsA.length, segmentsB.length);

    // order parent folder by name
    for (let i = 0; i < minLength - 1; i++) {
      const comparison = compareString(segmentsA[i], segmentsB[i]);

      if (comparison !== 0) {
        return comparison;
      }
    }

    // if they have a different length, folder get shown first
    if (segmentsA.length > minLength) {
      return -1;
    }

    if (segmentsB.length > minLength) {
      return 1;
    }

    return compareString(segmentsA[minLength - 1], segmentsB[minLength - 1]);
  });
```

where `compareString` is simply defined as:

```ts
function compareString(a: string, b: string) {
  if (a < b) {
    return -1;
  }

  if (a > b) {
    return 1;
  }

  return 0;
}
```

Reply:
Yup sounds good! Though for this to work, we'll need to be able to identify folders from files. Right now this sorting function get confused about empty folders:

[tk-filetree-sort.webm](https://github.com/user-attachments/assets/2087e758-e2da-4442-8ccf-25aee658b407)

Reply:
Oh good point, when I wrote that, I forgot we now have empty folders

Reply:
This ended up being a bit more complicated now that files and folders can be mixed there but I think I got it! Also added bunch of unit tests for this.